ui/dashboards/src/components/EmptyDashboard/EmptyDashboard.tsx

```diff
@@ -35,50 +37,90 @@ export interface EmptyDashboardProps {
-  </>
-);
+
+const DEFAULT_DESCRIPTION = {
+  edit: 'We currently support time series charts, gauge charts, stat charts and more!',
+  view: 'This dashboard is currently empty. Get Started by clicking the edit button.',
+};
 
 // Constants from specifics in designs to make the default messaging look good.
 const CONTAINER_WIDTH = '450px';
 const PRIMARY_CONTENT_WIDTH = '289px';
 
+const COMMON_BUTTON_PROPS = {
+  variant: 'outlined',
+  color: 'secondary',
+} as const;
+
+type EmptyDashboardActionsProps = Pick<EmptyDashboardProps, 'actions' | 'onEditButtonClick'> & {
+  isEditMode: boolean;
+};
+
+const EmptyDashboardActions = ({ actions, isEditMode, onEditButtonClick }: EmptyDashboardActionsProps) => {
+  if (actions && typeof actions !== 'boolean') {
+    // Custom actions
+    return actions;
+  }
+
+  if (actions === false) {
+    // Disable default actions
+    return null;
+  }
+
+  if (isEditMode) {
+    // Default edit mode actions
+    return (
+      <>
+        <AddPanelButton variant="outlined" color="secondary" label="Add Panel" fullWidth={true} />
+        <EditVariablesButton variant="outlined" color="secondary" label="Add Variables" fullWidth={true} />
+      </>
+    );
+  }
+
+  if (onEditButtonClick) {
+    // Default view mode actions
+    return <EditDashboardButton {...COMMON_BUTTON_PROPS} label="Edit Dashboard" onClick={onEditButtonClick} />;
+  }
+
+  return null;
+};
+
 /**
  * Communicate that a dashboard is empty and prompt the user to get started.
  */
 export const EmptyDashboard = ({
   title = DEFAULT_TITLE,
-  description = DEFAULT_DESCRIPTION,
-  actions,
+  description,
   additionalText,
+  ...otherProps
 }: EmptyDashboardProps) => {
-  // Show actions if it is truthy or if it is undefined. This allows the
-  // `undefined` case to fall back to the default while retaining the ability to
-  // set `false` to disable the actions entirely.
-  const showActions = !!actions || typeof actions === 'undefined';
+  const { isEditMode } = useEditMode();
+
+  const defaultDescription = isEditMode ? DEFAULT_DESCRIPTION.edit : DEFAULT_DESCRIPTION.view;
+  const actionsContent = <EmptyDashboardActions {...otherProps} isEditMode={isEditMode} />;
 
   return (
     <Box sx={{ width: CONTAINER_WIDTH, textAlign: 'center', margin: '0 auto' }}>
       <Box sx={{ width: PRIMARY_CONTENT_WIDTH, margin: '0 auto' }}>
         <Typography variant="h2" gutterBottom>
           {title}
         </Typography>
-        <Typography variant="body1">{description}</Typography>
-        {showActions && (
-          <Box sx={{ display: 'flex', gap: 2, marginTop: 1, justifyContent: 'center' }}>
-            {actions || DEFAULT_ACTIONS}
-          </Box>
+        <Typography variant="body1">{description ?? defaultDescription}</Typography>
+        {actionsContent && (
+          <Box sx={{ display: 'flex', gap: 2, marginTop: 1, justifyContent: 'center' }}>{actionsContent}</Box>
```

Comment:
You may have tried this and it wasn't any prettier, but sometime with things that vary by edit mode I've found it nice to do:

`isEditMode ? /* edit mode thing spelled out */ : /* view mode thing spelled out */` (probably all inside the container, which doesn't change) 

It'd be up to you if that actually made it any simpler

Reply:
The actions case has more complex logic than a simple ternary. That's why I pulled it out into a helper component for readability.